### PR TITLE
Fix/ADF-309/Do not remove classes due to restricted resources while deletion

### PR DESCRIPTION
**Pull request by shpran**
# [ADF-309](https://oat-sa.atlassian.net/browse/ADF-309)

## Require PRs
* https://github.com/oat-sa/tao-core/pull/3233

## Tasks
- [ ] Update `tao-core` dependency

### CODE REVIEW

**Review by hectoras — APPROVED** [review]

File: composer.json [hunk 1]
```diff
@@ -63,7 +63,7 @@
     "require": {
         "oat-sa/oatbox-extension-installer": "~1.1||dev-master",
         "oat-sa/generis": ">=14.0.0",
-        "oat-sa/tao-core": ">=48.45.0",
+        "oat-sa/tao-core": "dev-fix/adf-309/check-instances-access-during-deletion as 48.58.0",
```
Comment (gabrielfs7) [inline composer.json:66]: - [ ] Do not merge before update. Just a reminder

File: actions/class.Tests.php [hunk 2]
```diff
@@ -184,17 +186,30 @@ public function delete()
 
         if ($instance->isClass()) {
             $class = $this->getClass($instance->getUri());
-            $success = $this->getClassService()->deleteClass($class);
+
+            $classDeleter = $this->getClassDeleter();
+
+            try {
+                $classDeleter->delete($class);
```
Comment (gabrielfs7) [inline actions/class.Tests.php:193]: Does that mean that for other resources (assets, groups, test-takers, etc) we will need to do that?

**Review by gabrielfs7 — COMMENTED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [ ] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful

File: actions/class.Tests.php (hunk 2 above)
Comment (shpran) [reply actions/class.Tests.php:193]: Nope 🙂 
I checked all necessary places, most of them use `delete` method from RDF controller

**Review by gabrielfs7 — APPROVED** [review]

File: actions/class.Tests.php [hunk 3]
```diff
@@ -184,17 +186,36 @@ public function delete()
...
+                : __('Unable to delete %s.', $label);
         }
 
-        $message = $success ? __('%s has been deleted.', $label) : __('Unable to delete %s.', $label);
-
         $this->returnJson([
             'success' => $success,
             'message' => $message,
-            'deleted' => $success
+            'msg' => $message,
```
Comment (gabrielfs7) [inline actions/class.Tests.php:217]: We should stop using `msg` in the FE.

**Comment by AntonNeviarouski** [comment]
QA verified

**Pull request merged**